### PR TITLE
feat(share_plus)!: Native share UI for Windows

### LINKED ISSUE #333: Share.shareFiles Windows.
shareFiles() has not been implemented on Windows.

When the exception was thrown, this was the stack: 
#0      ShareWindows.shareFiles (package:share_plus_windows/share_plus_windows.dart:48:5)
#1      Share.shareFiles (package:share_plus/share_plus.dart:93:22)

### LINKED ISSUE #999: [Request]: Add support for the Windows Share Dialog
### Plugin

share_plus

### Use case

I am currently working on a multiplatform app and I wanted to implement a share function for them and I run into a missing feature.

### Proposal

Add support for the ability to share files/text via the Windows Share Dialog.

image

https://docs.microsoft.com/en-us/windows/apps/get-started/make-apps-great-for-windows#share-dialog

### PR DESCRIPTION
**Pull request by alexmercerind**
## Description

This brings support for all the existing `package:share_plus` methods to Windows:
- `share`
- `shareFiles`
- `shareWithResult`
- `shareFilesWithResult`
- `shareXFiles`

This pull request adds the modern Share UI (_i.e._ `DataTransferManager`) for Windows 10 or greater.
It is very rich in experience & uses same UWP API through WRL.

The feature parity is equivalent as other platforms. Text, subject, images & files are supported.

On older Windows versions, existing behavior is kept (opening `mailto:` URL with subject & body).

## Demo

https://user-images.githubusercontent.com/28951144/193889028-b15f6c3c-9fc1-442c-b472-39b848fcf0ae.mp4

## Related Issues

This PR closes following issues:

- Closes #333. 
- Closes #999.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I **haven't** updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by alexmercerind** [comment]
This is not exactly _a breaking change_.
But, since the change causes a big behavioural change from existing, I think it's best to ship it in a major version increment.

**Comment by alexmercerind** [comment]
Adding integration tests & `withResult` implementation.

**Comment by CodeDoctorDE** [comment]
The latest release uses XFiles, does it support it?

**Comment by alexmercerind** [comment]
> The latest release uses XFiles, does it support it?

Web specific.

**Comment by CodeDoctorDE** [comment]
Yeah, because of that. But shareFiles is now unsupported

**Comment by alexmercerind** [comment]
> Yeah, because of that. But shareFiles is now unsupported

It still uses `shareFilesWithResult` internally. No changes have been introduced to native code of Android, iOS, macOS or Linux.

This Windows implementation is equivalent & still working similarly.

https://github.com/fluttercommunity/plus_plugins/blob/69fbff0248fa2e6f6baf20fd57575ad047f9aea4/packages/share_plus/share_plus_platform_interface/lib/method_channel/method_channel_share.dart#L135-L152

**Comment by miquelbeltran** [comment]
Thanks for taking care of the changes in main @alexmercerind !

**Comment by alexmercerind** [comment]
If version `4.5.0` is decided to be merged into `5.0.0`, that's something we can do in `CHANGELOG`.

https://github.com/fluttercommunity/plus_plugins/blob/0d5069db5c4f508f3c995085b101337cc51d7238/packages/share_plus/share_plus/CHANGELOG.md?plain=1#L1-L11

**Review by miquelbeltran — COMMENTED** [review]
Tentatively looks good :) But I am not a Windows dev so I cannot comment, any devs familiar with Windows want to give it a look and comment?

**Comment by miquelbeltran** [comment]
Hey! Due to changes in our contributor guidelines, all PRs should now:

- Have a title that follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/) format. e.g. `feat(package_name)!: feature description`
- **DO NOT** modify the CHANGELOG.md or the version in the `pubspec.yaml`. This is going to be an automated process now.

We would ask you kindly to update the PR following these changes.

Thanks!

**Comment by alexmercerind** [comment]
Hi!
I've removed any version changes in `pubspec.yaml` & edits to `CHANGELOG.md`.
Also, I've rebased & cleaned-up commit history to keep things tidy.

**Comment by miquelbeltran** [comment]
Later today I will check this on my Windows machine

**Comment by miquelbeltran** [comment]
I checked this locally and works flawlessly, let's go!

**Review by miquelbeltran — APPROVED** [review]

File: packages/share_plus/share_plus_windows/test/share_plus_windows_test.dart [hunk 1]
```diff
@@ -1,40 +1,72 @@
...
 
 void main() {
-  test('registered instance', () {
-    ShareWindows.registerWith();
-    expect(SharePlatform.instance, isA<ShareWindows>());
-  });
-  test('url encoding is correct for &', () async {
-    final mock = MockUrlLauncherPlatform();
-    UrlLauncherPlatform.instance = mock;
+  test(
```
Comment (miquelbeltran) [inline packages/share_plus/share_plus_windows/test/share_plus_windows_test.dart:11]: Thanks a lot for including unit tests as well!

**Comment by miquelbeltran** [comment]
Thanks for the work @alexmercerind !

**Pull request merged**